### ISSUE TITLE
Release JOLT-SQL artifacts (code, models) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [Songjw133/Joint-Loss-Tuning-of-Text-to-SQL#2]
Hi @Songjw133 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I saw in your GitHub README (https://github.com/Songjw133/Joint-Loss-Tuning-of-Text-to-SQL) that you are "in the process of organizing the full code for release." That's great! Once your JOLT-SQL code and any fine-tuned model checkpoints (e.g., your models based on Qwen2.5-Coder) are ready, it would be fantastic to host them on the 🤗 Hub (https://huggingface.co/models) to improve their discoverability and visibility. We can add tags so that people find them when filtering, and link them directly to your paper page.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, if JOLT-SQL produces specific fine-tuned checkpoints, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

Let me know if you're interested in discussing this further or need any guidance as you prepare for release!

Cheers,

Niels
ML Engineer @ HF 🤗
